tools/who_what_benchmark/whowhatbench/model_loaders.py

```diff
@@ -0,0 +1,252 @@
+            model = OVModelForCausalLM.from_pretrained(
+                model_id,
+                config=config,
+                trust_remote_code=True,
+                use_cache=True,
+                device=device,
+                ov_config=ov_config,
+            )
+
+    return model
+
+
+def load_text2image_genai_pipeline(model_dir, device="CPU", ov_config=None):
+    try:
+        import openvino_genai
+    except ImportError:
+        logger.error(
+            "Failed to import openvino_genai package. Please install it.")
+        exit(-1)
+
+    return GenAIModelWrapper(
+        openvino_genai.Text2ImagePipeline(model_dir, device=device, **ov_config),
+        model_dir,
+        "text-to-image"
+    )
+
+
+def load_text2image_model(
+    model_id, device="CPU", ov_config=None, use_hf=False, use_genai=False
+):
+    if use_genai:
+        logger.info("Using OpenvINO GenAI API")
+        model = load_text2image_genai_pipeline(model_id, device, ov_config)
+    elif use_hf:
+        logger.info("Using HF Transformers API")
+        model = DiffusionPipeline.from_pretrained(
+            model_id, trust_remote_code=True)
+    else:
+        logger.info("Using Optimum API")
+        from optimum.intel import OVPipelineForText2Image
+        TEXT2IMAGEPipeline = OVPipelineForText2Image
+
+        try:
+            model = TEXT2IMAGEPipeline.from_pretrained(
+                model_id, trust_remote_code=True, device=device, ov_config=ov_config
+            )
+        except ValueError:
+            config = AutoConfig.from_pretrained(
+                model_id, trust_remote_code=True)
+            model = TEXT2IMAGEPipeline.from_pretrained(
+                model_id,
+                config=config,
+                trust_remote_code=True,
+                use_cache=True,
+                device=device,
+                ov_config=ov_config,
+            )
+
+    return model
+
+
+def load_visual_text_genai_pipeline(model_dir, device="CPU", ov_config=None):
+    try:
+        import openvino_genai
+    except ImportError as e:
+        logger.error("Failed to import openvino_genai package. Please install it. Details:\n", e)
+        exit(-1)
+
+    return GenAIModelWrapper(
+        openvino_genai.VLMPipeline(model_dir, device, **ov_config),
+        model_dir,
+        "visual-text"
+    )
+
+
+def load_visual_text_model(
+    model_id, device="CPU", ov_config=None, use_hf=False, use_genai=False
+):
+    if use_hf:
+        logger.info("Using HF Transformers API")
+        config = AutoConfig.from_pretrained(model_id, trust_remote_code=True)
+        try:
+            model = AutoModelForVision2Seq.from_pretrained(
+                model_id, trust_remote_code=True, device_map=device.lower()
+            )
+        except ValueError:
+            try:
+                model = AutoModel.from_pretrained(
+                    model_id, trust_remote_code=True, device_map=device.lower()
+                )
+            except ValueError:
```

Comment:
I think it may be more robust if you will use model_type from config for selection model loading approach (e.g. logic bellow required only for phi3-vision, but there is no guarantee that some other model will not go by this path)

Reply:
Right, and as you can see this is the last clause in the try-except chain. Anyway `load_visual_text_model` is called if "visual-text" model_type is passed explicitly.

Reply:
To tell the truth, I like it more rather than hardcode `model_type == "phi3_v"`